File: js/src/record/checkSolRecord.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.checkSolRecord = void 0;
const ed25519_1 = require("@noble/curves/ed25519");
/**
 * This function can be used to verify the validity of a SOL record
 * @param record The record data to verify
 * @param signedRecord The signed data
 * @param pubkey The public key of the signer
 * @returns
 */
const checkSolRecord = (record, signedRecord, pubkey) => {
    return ed25519_1.ed25519.verify(signedRecord, record, pubkey.toBytes());
};
exports.checkSolRecord = checkSolRecord;

File: python/record/check_sol_record.py
from solders.pubkey import Pubkey
from nacl.signing import VerifyKey


def check_sol_record(
        record: bytes,
        signed_record: bytes,
        pubkey: Pubkey,
) -> bool:
    """
    Check if a SOL record is valid, on ed25519 curve.
    :param record: The record to check.
    :param signed_record: The signed record to check.
    :param pubkey: The pubkey to check against.
    :return: True if the record is valid, False otherwise.
    """
    try:
        verify_key = VerifyKey(bytes(pubkey))
        verify_key.verify(record, signed_record)
        return True  # Signature is valid
    except Exception:
        return False  # Signature is invalid